nearcore/src/migrations.rs

```diff
@@ -38,8 +47,97 @@ impl<'a> near_store::StoreMigrator for Migrator<'a> {
                 &self.config.genesis.config,
                 &self.config.config.store,
             ),
-            47 => Ok(()), // TODO(continuous_epoch_sync): Implement the migration
+            47 => {
+                migrate_47_to_48(hot_store, self.config.genesis.config.transaction_validity_period)
+            }
             DB_VERSION.. => unreachable!(),
         }
     }
 }
+
+/// This migration does two things:
+/// 1. Generate and save the compressed epoch sync proof
+/// 2. Clear the block headers from genesis to tail in hot_store
+fn migrate_47_to_48(
+    hot_store: &Store,
+    transaction_validity_period: BlockHeightDelta,
+) -> anyhow::Result<()> {
+    tracing::info!(target: "migrations", "starting migration from DB version 47 to 48");
+    update_epoch_sync_proof(hot_store.clone(), transaction_validity_period)?;
+    delete_old_block_headers(hot_store)?;
+    Ok(())
+}
+
+fn update_epoch_sync_proof(
+    store: Store,
+    transaction_validity_period: BlockHeightDelta,
+) -> anyhow::Result<()> {
+    let epoch_store = store.epoch_store();
+    let chain_store = store.chain_store();
+
+    tracing::info!(target: "migrations", "updating existing epoch sync proof to compressed format");
+
+    // First we move any existing epoch sync proof to the compressed format
+    // Note that while accessing the proof, we need to read directory from DBCol::EpochSyncProof
+    // as we can't use the epoch_store.get_epoch_sync_proof() method due to
+    // ProtocolFeature::ContinuousEpochSync being enabled
+    if let Some(proof) = store.get_ser::<EpochSyncProof>(DBCol::EpochSyncProof, &[]).unwrap() {
```

Comment:
See https://github.com/near/nearcore/pull/14738/files `get_epoch_sync_proof` for why we are doing this.